### PR TITLE
Centralize dependency versions for client libraries

### PR DESCRIPTION
**Pull request by bsiegel**
The management libraries will be tackled in a separate PR.

### CODE REVIEW

File: sdk/eventhub/Microsoft.Azure.EventHubs.Processor/src/Microsoft.Azure.EventHubs.Processor.csproj [hunk 1]
```diff
@@ -17,17 +17,17 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Newtonsoft.Json" Version="12.0.1" />
-    <PackageReference Include="Microsoft.Azure.KeyVault.Core" Version="3.0.3" />
+    <PackageReference Include="Newtonsoft.Json" VersionOverride="12.0.1" />
```
Comment (weshaggard) [inline sdk/eventhub/Microsoft.Azure.EventHubs.Processor/src/Microsoft.Azure.EventHubs.Processor.csproj]: For this one lets try and pull it down the 10.0.3

File: Packages.props [hunk 2]
```diff
@@ -0,0 +1,108 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <ItemGroup Condition=" '$(TargetFramework)' == 'netstandard1.4' ">
+    <PackageReference Update="Microsoft.AspNetCore.WebUtilities" Version="1.1.2" />
+  </ItemGroup>
+  <ItemGroup Condition=" '$(TargetFramework)' == 'netstandard2.0' ">
+    <PackageReference Update="Microsoft.AspNetCore.WebUtilities" Version="2.2.0" />
+  </ItemGroup>
+  <!-- <ItemGroup Condition="'$(IsTargetingNetFx)' == 'true'">
```
Comment (weshaggard) [inline Packages.props]: Let's set Newtonsoft.Json to version 10.0.3 as the global default and add overrides if they are needed. I think we should try to move the 12.0.1 back to 10.0.3 but there may be some cases were we need to stick to 6.0.8 for the older target frameworks but that would be an override if needed.

File: Packages.props [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    <PackageReference Update="Microsoft.Extensions.Configuration" Version="1.0.2" />
+    <PackageReference Update="Microsoft.Extensions.Configuration.Abstractions" Version="2.2.0" />
+    <PackageReference Update="Microsoft.Extensions.Configuration.Binder" Version="2.2.0" />
+    <PackageReference Update="Microsoft.Extensions.Configuration.Json" Version="1.0.2" />
+    <PackageReference Update="Microsoft.Extensions.DependencyInjection.Abstractions" Version="2.2.0" />
+    <PackageReference Update="Microsoft.Extensions.DependencyInjection" Version="2.2.0" />
+    <PackageReference Update="Microsoft.Extensions.Logging.Abstractions" Version="2.2.0" />
+    <PackageReference Update="Microsoft.Extensions.Options" Version="2.2.0" />
+    <PackageReference Update="Microsoft.Extensions.PlatformAbstractions" Version="1.1.0" />
+    <!-- <PackageReference Update="Microsoft.IdentityModel.Clients.ActiveDirectory" Version="3.14.0" /> -->
```
Comment (weshaggard) [inline Packages.props]: Let's try aligning everyone to the most common version which appears to be 4.5.1.

File: Packages.props [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    <PackageReference Update="System.Runtime.InteropServices" Version="4.3.0" />
+    <PackageReference Update="System.Runtime.Serialization.Primitives" Version="4.3.0" />
+    <PackageReference Update="System.Security.Cryptography.Algorithms" Version="4.3.0" />
+    <PackageReference Update="System.Security.Cryptography.Cng" Version="4.3.0" />
+    <PackageReference Update="System.Security.Cryptography.Primitives" Version="4.3.0" />
+    <PackageReference Update="System.Text.Json" Version="4.6.0-preview6.19259.10" />
+    <PackageReference Update="System.Threading.Tasks.Extensions" Version="4.5.1" />
+    <PackageReference Update="System.ValueTuple" Version="4.5.0" />
+    <PackageReference Update="WindowsAzure.ServiceBus" Version="5.1.0" />
+    <!-- <PackageReference Update="WindowsAzure.Storage" Version="8.1.4" /> -->
```
Comment (weshaggard) [inline Packages.props]: Lets try to align everyone to 8.1.4.

**Comment by weshaggard** [comment]
Looks like EventHub seems to be the most common outlier in the version differences lets try and use the common version there as well and see if we can get it working and if not we consider an override or upgrade.

File: Packages.props [hunk 5]
```diff
@@ -0,0 +1,103 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (weshaggard) [inline Packages.props]: I would like to try and keep the root as clean as possible so prefer this to go into the eng directory.  According to the docs at https://github.com/Microsoft/MSBuildSdks/tree/master/src/CentralPackageVersions you can set the `CentralPackagesFile` in the props file to put the packages file somewhere else.

**Comment by mitchdenny** [comment]
Large cross repo commit == abuse errors going off ... looks like we still need the token caching solution to go live :)

**Comment by bsiegel** [comment]
@weshaggard Everything seems to be working here with the exception of a single failing eventhubs test. I have done a bit of searching and I'm not sure how to fix this - any suggestions? The failure is:

```
Failed   DisposeAsyncDelegatesToCloseAsync

Error Message:
 System.IO.FileLoadException : Could not load file or assembly 'System.Runtime.CompilerServices.Unsafe, Version=4.0.4.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
Stack Trace:
   at System.Threading.Tasks.ValueTask.get_IsCompleted()
   at System.Runtime.CompilerServices.ValueTaskAwaiter.get_IsCompleted()
   at Azure.Messaging.EventHubs.Tests.EventHubClientTests.<DisposeAsyncDelegatesToCloseAsync>d__25.MoveNext() in D:\a\1\s\sdk\eventhub\Azure.Messaging.EventHubs\tests\EventHubClient\EventHubClientTests.cs:line 537
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NUnit.Framework.Internal.TaskAwaitAdapter.GenericAdapter`1.GetResult()
   at NUnit.Framework.Internal.AsyncToSyncAdapter.Await(Func`1 invoke)
   at NUnit.Framework.Internal.Commands.TestMethodCommand.RunTestMethod(TestExecutionContext context)
   at NUnit.Framework.Internal.Commands.TestMethodCommand.Execute(TestExecutionContext context)
   at NUnit.Framework.Internal.Execution.SimpleWorkItem.PerformWork()
```

File: eng/Packages.Data.props [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    <PackageReference Update="System.Runtime.CompilerServices.Unsafe" Version="4.5.2" />
+    <PackageReference Update="System.Runtime.Extensions" Version="4.3.0" />
+    <PackageReference Update="System.Runtime.InteropServices" Version="4.3.0" />
+    <PackageReference Update="System.Runtime.Serialization.Primitives" Version="4.3.0" />
+    <PackageReference Update="System.Runtime" Version="4.3.0" />
+    <PackageReference Update="System.Security.Cryptography.Algorithms" Version="4.3.0" />
+    <PackageReference Update="System.Security.Cryptography.Cng" Version="4.3.0" />
+    <PackageReference Update="System.Security.Cryptography.Primitives" Version="4.3.0" />
+    <PackageReference Update="System.Text.Json" Version="4.6.0-preview6.19259.10" />
+    <PackageReference Update="System.Threading.Tasks.Extensions" Version="4.5.1" />
```
Comment (weshaggard) [inline eng/Packages.Data.props]: I believe this is what is causing your test failure. You need to bump System.Threading.Tasks.Extensions to 4.5.2.

**Comment by bsiegel** [comment]
In case anyone comes in from the internet searching for that exception stack trace above, the solution is that the following libraries must all use the **SAME** version (4.5.1 or 4.5.2, it most likely doesn't matter as long as they all use the same version):

- System.Memory
- System.Runtime.CompilerServices.Unsafe
- System.Threading.Tasks.Extensions

**Review by weshaggard — APPROVED** [review]

**Pull request merged**